agent/src/beerocks/slave/gate/1905_beacon_query_to_vs.cpp

```diff
@@ -0,0 +1,113 @@
+
+#include "1905_beacon_query_to_vs.h"
+#include <bcl/beerocks_defines.h>
+#include <mapf/common/utils.h>
+
+namespace gate {
+
+bool load(std::shared_ptr<beerocks_message::cACTION_MONITOR_CLIENT_BEACON_11K_REQUEST> lhs,
+          const ieee1905_1::CmduMessageRx &rhs)
+{
+    // escape I
+    if (!lhs) {
+        LOG(ERROR) << "can't fill empty pointer. lhs is nullptr. returning";
+        return false;
+    }
+
+    // escape II
+    // get the correct type of the message
+    auto beacon_metrics_query = rhs.getClass<wfa_map::tlvBeaconMetricsQuery>();
+    if (!beacon_metrics_query) {
+        LOG(ERROR) << "tlvBeaconMetricsQuery is missing - can't read from invalid input";
+        return false;
+    }
+
+    auto &lhs_params = lhs->params();
+
+    lhs_params.bssid   = beacon_metrics_query->bssid();
+    lhs_params.channel = beacon_metrics_query->channel_number();
+
+    lhs_params.measurement_mode = beerocks::MEASURE_MODE_ACTIVE;
+    lhs_params.duration         = beerocks::BEACON_MEASURE_DEFAULT_ACTIVE_DURATION;
+
+    lhs_params.expected_reports_count = 1;
+
+    lhs_params.rand_ival = beerocks::BEACON_MEASURE_DEFAULT_RANDOMIZATION_INTERVAL;
+    lhs_params.sta_mac   = beacon_metrics_query->associated_sta_mac();
+
+    lhs_params.op_class = beacon_metrics_query->operating_class();
+
+    // FixMe: not sure about this
+    lhs_params.report = beacon_metrics_query->reporting_detail_value();
+
+    // values based on https://github.com/prplfoundation/prplMesh/pull/1114#discussion_r406326546
+    lhs_params.repeats            = 0;
+    lhs_params.parallel           = 0;
+    lhs_params.enable             = 0;
+    lhs_params.request            = 1;
+    lhs_params.mandatory_duration = 0;
+    lhs_params.use_optional_ssid  = 0;
+
+    mapf::utils::copy_string(lhs_params.ssid, beacon_metrics_query->ssid_str().c_str(),
+                             beerocks::message::WIFI_SSID_MAX_LENGTH);
+
+    // how many elements in the list of channels, we support only 1 at the moment
+    auto ap_ch_report_length = beacon_metrics_query->ap_channel_reports_list_length();
+
+    // escape III - data validation
+    // ap_ch_report_length != 0 ==> channel_number == 255
+    // we verify with the already extracted values (we could do it withthe original as well)
+    if (ap_ch_report_length != 0 && lhs_params.channel != 255) {
+        LOG(ERROR) << "inconsistency between channel report length and channel number. please take "
```

Comment:
(nitpick) The message comes from the outside, so referring to the spec is of limited value.

Reply:
I don't understand the comment.
The reason I am referring to the specification is because this is the _sole_ reason the code is written the way it is. the idea is to save the burden from a future developer that will try to figure out why this code was written this way, and what does this error log mean.

Reply:
Hm, I didn't word that very well... What I mean is: if we get invalid values here, it's not a programming error, but it's invalid input. Referring to the specification in a comment makes complete sense, but not so much in the log. The log is going to be looked at by a system administrator, not by a prplmesh programmer. Hm... the more I'm arguing about this, the more it looks like I'm wrong - it *is* actually useful to refer to the spec so the sysadmin can debug the device that was sending this invalid message...